### DISCUSSION TITLE
Customise the json request body

### DISCUSSION THREAD

**Discussion by sagarnayak** [rwf2/Rocket#1763] [Questions]
I want to handle any unwanted data types for a struct when I get a JSON request in a post req. If I get a u16 in place of a String then the only place I can handle this is a [catcher](https://github.com/sagarnayak/RustTestWebApp/blob/ef3278e3db18b7cf97cc2e8f3ea4dab44130cf2d/src/controllers/error_handlers.rs#L9).

also, is there a way to customize the field name in [Struct](https://github.com/sagarnayak/RustTestWebApp/blob/ef3278e3db18b7cf97cc2e8f3ea4dab44130cf2d/src/model/password_verification_req.rs#L4) for JSON as it is possible in [Form data](https://rocket.rs/v0.5-rc/guide/requests/#field-renaming)?

Current implementation - [Github Repo](https://github.com/sagarnayak/RustTestWebApp/blob/ef3278e3db18b7cf97cc2e8f3ea4dab44130cf2d/src/controllers/passwords.rs#L38)

**Comment by SergioBenitez** [comment]
I don't understand the question. Is this different from https://github.com/SergioBenitez/Rocket/discussions/1757? For serde, see serde.rs.

**Reply by sagarnayak** [reply]
Yes, I think it's different.

The case you referred handles any unexpected values in the path param.

But what I want is any unexpected value in a struct that I am getting in a post as a body.

	#[post("/verifyHashedMyPassword", data = "<password_verification_request>")]
	pub fn verify_hashed_password(
		password_verification_request: Json<PasswordVerificationRequest>,
	)

the struct is -

	use serde::{Deserialize, Serialize};

	#[derive(Serialize, Deserialize, Debug)]
	pub struct PasswordVerificationRequest {
		pub original_password: String,
		pub hash: String,
	}

Now if a user sends an i32 in place of hash. I want to send a specific response that you have entered a number in the hash. please send a string.

I think this explanation would make my case clear.
